### ISSUE TITLE
Session recording do not work with hashed assets

### ISSUE THREAD

**Issue by hodbn** [PostHog/posthog#17865]
## Bug description

According to [docs](https://posthog.com/docs/session-replay/troubleshooting#ensure-your-website-permits-posthog-cors), most of the time the session recording feature will fetch assets (fonts, js, images, css, etc.) inline - when the recording is captured.

But for some assets, it will only access the resource in "real-time" - when rendering a recording.

If a "hashed asset" (e.g. `https://example.com/assets/vendor-cd39faaa.js`) has been rebuilt in the timeframe between the recording capture to its rendering, the asset won't load when rendering the recording.

## How to reproduce

1. Create an empty [vite app](https://vitejs.dev/guide/) (or any other bundler that uses hashed assets).
2. Install the react posthog snippet in the app.
3. Deploy the app on Netlify (for example, any other service that uses [atomic deploys](https://docs.netlify.com/site-deploys/overview/#) will do).
4. Record a session.
5. Rebuild and re-deploy the app, the URLs of the hashed assets are now changed.
6. Replay the session from step `4`, you'll notice that some requests (e.g., `vendor-XXXXXXXX.js` will point to 404 URLs.

## Environment

- [x] PostHog Cloud US, project ID: 34026
- [ ] PostHog Cloud EU, project ID: None
- [ ] PostHog Hobby self-hosted with `docker compose`, version/commit: None
- [ ] PostHog self-hosted with Kubernetes (deprecated, see ["Sunsetting Kubernetes support"](https://posthog.com/blog/sunsetting-helm-support-posthog)), version/commit: None

## Additional context

None.

**Comment by benjackwhite** [comment]
I'm wondering - is any of this stuff actually breaking? We don't use js assets at all for the replay...

**Comment by hodbn** [comment]
@benjackwhite You're right, I can't link the "broken stuff" I see directly to the failed `vendor-XXXXX.js` network requests. I understand from what you're saying that the requests are failing but that the replay does not rely on them and it's consistent with what I see (JS is working).

Right now, some images won't load and I can see their failing network requests too.
Is this the reason they're not loading?

**Comment by rotimi-best** [comment]
I think I am facing a similar issue.

Here is a screen recording: https://eu.posthog.com/shared/bB3dp9zMuxxkKw5pNBiBMQvDGWhHoA
